biz/auth.manager.js

```diff
@@ -80,10 +56,77 @@ class AuthManager extends BaseManager {
         throw new NotFound(MSG.USER_NOT_FOUND);
       }
       throw new ValidationError(MSG.VALIDATION_ERROR, validationResult.errors);
+    } catch(err) {
+      throw err;
+    }
+  }
+
+  async signUp(bodyParams, SCHEMA, model) {
+    try {
+      const validationResult = this.validate(SCHEMA, bodyParams);
+      if (validationResult.valid) {
+        let { phone_number, email_id, password } = bodyParams;
+
+        const checkExist = await this._authRepository.findOne(
+          model,
+          phone_number,
+          email_id
+        );
+
+        if (!checkExist) {
+          bodyParams.customer_id = randomize("Aa0", 5);
+          bodyParams.password = await bcrypt.hash(password, saltRounds);
+          const saveCustomerData = await this._authRepository.saveOne(
+            model,
+            bodyParams
+          );
+          return saveCustomerData;
+        }
+        throw new DuplicateError(MSG.DUPLICATE_USER);
+      }
+      throw new ValidationError(MSG.VALIDATION_ERROR, validationResult.errors);
     } catch (err) {
       throw err;
     }
   }
+
+  async addService(bodyParams, model){
+    try{
+      const validationResult = this.validate(SCHEMA.SERVICES);
+      if(validationResult.valid){
+        bodyParams.service_id = randomize("Aa0", 4);
+        const serviceAdded = await this._authRepository.saveOne(
+          model
+        );
+        return serviceAdded;
+      }
+      throw new ValidationError(MSG.VALIDATION_ERROR)
+    }catch (err){
+      throw err;
+    }
+  }
+
+  async updateService(bodyParams, model){
+    try{
+      let { service_id, name, description, cost } = bodyParams;
+      const checkExist = await this._authRepository.findOne(
+        service_id
+      );
+      if(checkExist){
+        model.findOneAndUpdate(service_id, {name, description, cost}, {new: true}, function(err, model){
```

Comment:
use auth repository for mongo functions. pass bodyparams in that function. Make a new function in that file